lib/src/line_splitting/line_splitter.dart

```diff
@@ -122,7 +119,7 @@ class LineSplitter {
   final _queue = SolveStateQueue();
 
   /// The lowest cost solution found so far.
-  SolveState _bestSolution;
+  SolveState? _bestSolution;
```

Comment:
Does this need to be a field?  It looks like it's only used in `apply`, and assuming that `apply` is only called once for each instance of `LineSplitter` (which I *think* is the case), we could make this a local variable and the benefit of flow analysis to promote it to non-nullable sometimes.

Reply:
Ah, good catch!